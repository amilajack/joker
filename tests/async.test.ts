import { jokerFixture } from '.';

describe('async joker', () => {
  it('should reject if main function is forcefully aborted', async () => {
    await expect(
      jokerFixture()
        .run('node timeout.js')
        .timeout(100)
        .end()
    ).rejects.toThrowErrorMatchingSnapshot();
  });

  it('should reject if assertion fails', async () => {
    await expect(
      jokerFixture()
        .run('echo foo')
        .stdout('baz')
        .end()
    ).rejects.toThrowErrorMatchingSnapshot();
  });
});
