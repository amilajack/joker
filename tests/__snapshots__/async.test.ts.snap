// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`async joker should reject if assertion fails 1`] = `"\`echo foo\`: Expected stdout to match \\"baz\\". Actual: \\"foo\\""`;

exports[`async joker should reject if assertion fails 2`] = `
"\`node -e \\"require('foo')\\"\`: Expected stderr to match \\"baz\\". Actual: \\"internal/modules/cjs/loader.js:615
    throw err;
    ^

Error: Cannot find module 'foo'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:613:15)
    at Function.Module._load (internal/modules/cjs/loader.js:539:25)
    at Module.require (internal/modules/cjs/loader.js:667:17)
    at require (internal/modules/cjs/helpers.js:20:18)
    at [eval]:1:1
    at Script.runInThisContext (vm.js:124:20)
    at Object.runInThisContext (vm.js:314:38)
    at Object.<anonymous> ([eval]-wrapper:9:26)
    at Module._compile (internal/modules/cjs/loader.js:738:30)
    at evalScript (internal/process/execution.js:60:25)\\""
`;

exports[`async joker should reject if main function is forcefully aborted 1`] = `"\`node timeout.js\`: Command execution terminated (timeout)"`;

exports[`async joker should show stack traces for failing js code 1`] = `undefined`;
